### PR TITLE
feat: Add HTML previews

### PR DESCRIPTION
**Pull request by ddeboer**
* Translate labels in preview based on browser’s Accept-Language header.

#265.﻿

### CODE REVIEW

File: src/reconciliation/preview.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      <p>${literal(term.scopeNotes)}</p>
+      <dl>
+        ${
+          term.altLabels.length > 0
+            ? `<dt>Alternatieve labels</dt><dd>${literal(term.altLabels)}</dd>`
+            : ''
+        }
+        ${relatedTerms('Bredere termen', term.broaderTerms)}
+        ${relatedTerms('Nauwere termen', term.narrowerTerms)}
+        ${relatedTerms('Gerelateerde termen', term.relatedTerms)}
```
Comment (ddeboer) [inline src/reconciliation/preview.ts:29]: Haven’t found a way in the Reconciliation spec to parameterise the language, so decided to use Dutch here. Would English make more sense?

File: src/server/server.ts [hunk 2]
```diff
@@ -37,6 +41,13 @@ export async function server(
   });
   server.register(fastifyCors);
   server.register(formBodyPlugin);
+  server.decorateRequest('previewUrl', '');
+  server.addHook('onRequest', (request, reply, done) => {
+    request.raw.previewUrl =
+      request.protocol + '://' + request.hostname + '/preview/{{id}}';
+    done();
+  });
```
Comment (ddeboer) [inline src/server/server.ts:49]: It seems the preview URL must be absolute, so we have to dynamically get the request’s hostname to build the URL.

File: src/reconciliation/preview.ts (hunk 1 above)
Comment (ddeboer) [reply src/reconciliation/preview.ts:29]: Good suggestion! 👍

OpenRefine indeed sends an `Accept-Language` header with preview requests. By inspecting the headers as they arrived at our preview service, I found that `Accept-Language`’s value depends not on the OpenRefine language setting but on the browser’s locale. While not ideal (Dutch users may have their browser set to English), we can respond to the value in our preview service.

Comment (sdevalk) [reply src/reconciliation/preview.ts:29]: Excellent!

**Pull request merged**